src/interpret.jl

```diff
@@ -130,25 +132,29 @@ function resolvefc(frame, @nospecialize(expr))
     error("unexpected ccall to ", expr)
 end
 
-function collect_args(frame::Frame, call_expr::Expr; isfc::Bool=false)
+function collect_args(@nospecialize(recurse), frame::Frame, call_expr::Expr; isfc::Bool=false)
     args = frame.framedata.callargs
     resize!(args, length(call_expr.args))
     mod = moduleof(frame)
     args[1] = isfc ? resolvefc(frame, call_expr.args[1]) : @lookup(mod, frame, call_expr.args[1])
     for i = 2:length(args)
-        args[i] = @lookup(mod, frame, call_expr.args[i])
+        if isexpr(call_expr.args[i], :call)
+            args[i] = lookup_or_eval(recurse, frame, call_expr.args[i])
```

Comment:
With this change, we might need to test
```julia
iscallexpr(ex::Expr) = ex.head === :call
@interpret iscallexpr(:(sin(3.14))
```
We need to be sure we distinguish a call expression intended as an argument from one produced by lowering.